Golem.Tests/Utils.cs

```diff
@@ -110,9 +113,71 @@ public async static Task<T> ReadChannel<T>(ChannelReader<T> channel, Func<T, boo
 
             throw new Exception($"Failed to find matching {nameof(T)} within {timeoutMs} ms.");
         }
+
+        public static Channel<GolemStatus> StatusChannel(Golem golem, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Golem, GolemStatus>(golem, nameof(IGolem.Status), loggerFactory);
+
+        public static Channel<Job?> JobChannel(Golem golem, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Golem, Job?>(golem, nameof(IGolem.CurrentJob), loggerFactory);
+
+        public static Channel<JobStatus> JobStatusChannel(Job job, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Job, JobStatus>(job, nameof(job.Status), loggerFactory);
+
+        public static Channel<GolemLib.Types.PaymentStatus?> JobPaymentStatusChannel(Job job, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Job, GolemLib.Types.PaymentStatus?>(job, nameof(job.PaymentStatus), loggerFactory);
+
+        public static Channel<GolemUsage?> JobUsageChannel(Job job, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Job, GolemUsage?>(job, nameof(job.CurrentUsage), loggerFactory);
+
+        public static Channel<List<Payment>?> JobPaymentConfirmationChannel(Job job, ILoggerFactory loggerFactory) =>
+            PropertyChangeChannel<Job, List<Payment>?>(job, nameof(job.PaymentConfirmation), loggerFactory);
+
+        public static void CheckPortIsAvailable()
+        {
+            if (!IsPortAvailable(EnvironmentBuilder.DefaultLocalAddress, EnvironmentBuilder.DefaultApiPort))
+            {
+                throw new Exception("API port is open");
+            }
+        }
+
+        public static bool IsPortAvailable(
+            string host,
+            int port,
+            int retries = 30,
+            int retry_delay = 1000,
+            int connection_timeout = 250
+        )
+        {
+            for (int retries_count = 0; retries_count < retries; retries_count++)
+            {
+                try
+                {
+                    var client = new TcpClient();
```

Comment:
You should use `using var client` otherwise you leak resources

Reply:
Fixed.